packages/e2e/cypress/e2e/api/organizationPermissions.cy.ts

```diff
@@ -210,137 +213,169 @@ describe('Lightdash API organization permission tests', () => {
+                    });
+                });
             });
         });
     });
 });
 
-describe('Lightdash API tests for interactive_viewer org user', () => {
-    let email;
-
-    before(() => {
-        cy.loginWithPermissions('interactive_viewer', []).then((e) => {
-            email = e;
-        });
-    });
-    beforeEach(() => {
-        cy.loginWithEmail(email);
-    });
-    it('Should identify user', () => {
-        cy.request(`${apiUrl}/user`).then((resp) => {
-            expect(resp.status).to.eq(200);
-            expect(resp.body.results).to.have.property('email', email);
-            expect(resp.body.results).to.have.property(
-                'role',
-                'interactive_viewer',
-            );
-        });
-    });
+describe('lightdash API tests for project creation permissions', () => {
+    [
+        {
+            role: 'admin',
+            canCreateProject: true,
+            canCreatePreview: true,
+        },
+        {
+            role: 'developer',
+            canCreateProject: false,
+            canCreatePreview: true,
+        },
+        {
+            role: 'editor',
+            canCreateProject: false,
+            canCreatePreview: false,
+        },
+        {
+            role: 'interactive_viewer',
+            canCreateProject: false,
+            canCreatePreview: false,
+        },
+        {
+            role: 'viewer',
+            canCreateProject: false,
+            canCreatePreview: false,
+        },
+        {
+            role: 'member',
+            canCreateProject: false,
+            canCreatePreview: false,
+        },
+    ].forEach(({ role, canCreatePreview, canCreateProject }) => {
+        describe(`org user with '${role}' role`, () => {
+            let email;
+
+            before(() => {
+                cy.loginWithPermissions(role, []).then((e) => {
+                    email = e;
+                });
+            });
 
-    it('Should get success response (200) from GET public endpoints', () => {
-        const endpoints = ['/livez', '/health', '/flash'];
-        endpoints.forEach((endpoint) => {
-            cy.request(`${apiUrl}${endpoint}`).then((resp) => {
-                expect(resp.status).to.eq(200);
-                expect(resp.body).to.have.property('status', 'ok');
+            beforeEach(() => {
+                cy.loginWithEmail(email);
             });
-        });
-    });
 
-    it('Should get success response (200) from GET org', () => {
-        const endpoint = `${apiUrl}/org/`;
+            it('should get a parameter error when sending POST to project with DEFAULT and an UPSTREAM', () => {
+                const endpoint = `${apiUrl}/org/projects/`;
 
-        cy.request({
-            url: endpoint,
-            headers: { 'Content-type': 'application/json' },
-            method: 'GET',
-        }).then((resp) => {
-            expect(resp.status).to.eq(200);
-        });
-    });
+                cy.request({
```

Comment:
is this creating `projects` for those who are allowed to create projects ?

Reply:
ahh, this is for DEFAULT and UPSTREAM, so it should faill even for admins, right ?

Reply:
correct!